src/code/z_viscvg.c

```diff
@@ -75,20 +109,29 @@ void func_800ACE98(struct_801664F0* this, Gfx** gfxp) {
     }
 
     switch (this->type) {
-        case 1:
-            gSPDisplayList(gfx++, D_8012AC40);
+        case FB_FILTER_CVG_RGB:
+            gSPDisplayList(gfx++, sCoverageRGBDL);
             break;
-        case 2:
-            gDPSetColor(gfx++, G_SETPRIMCOLOR, this->color.rgba);
-            gSPDisplayList(gfx++, D_8012AC58);
+
+        case FB_FILTER_CVG_RGB_PRIMFOG:
+            // Set primitive color for uniform fogging in custom RenderMode
+            gDPSetColor(gfx++, G_SETPRIMCOLOR, this->primColor.rgba);
+            gSPDisplayList(gfx++, sCoverageRGBPrimFogDL);
             break;
-        case 3:
-            gDPSetColor(gfx++, G_SETBLENDCOLOR, this->color.rgba);
-            gSPDisplayList(gfx++, D_8012AC00);
+
+        case FB_FILTER_CVG:
+            // Set background color for G_RM_VISCVG
+            gDPSetColor(gfx++, G_SETBLENDCOLOR, this->primColor.rgba);
```

Comment:
`VisCvg.primColor` is used as prim/blend/fog color so the previous name "color" is probably better

Reply:
I did it for consistency with the other two Vis files, don't think it improves it to name it `color`, especially since there's an envColor too.

Reply:
Well it's simply not a primColor in some cases. You can name it otherColor if you want to make it stand out from envColor? Or mainColor ...?

Reply:
Come to think of it
```c
typedef struct {
    /* 0x00 */ u32 type;
    /* 0x04 */ u32 setScissor;
    /* 0x08 */ Color_RGBA8_u32 primColor;
    /* 0x0C */ Color_RGBA8_u32 envColor; // unused
} VisCvg; // size = 0x10

typedef struct {
    /* 0x00 */ u32 unk_00;
    /* 0x04 */ u32 setScissor;
    /* 0x08 */ Color_RGBA8_u32 primColor;
    /* 0x0C */ Color_RGBA8_u32 envColor;
    /* 0x10 */ u16* tlut;
    /* 0x14 */ Gfx* dList;
} VisMono; // size = 0x18

typedef struct {
    /* 0x00 */ u32 useRgba;
    /* 0x04 */ u32 setScissor;
    /* 0x08 */ Color_RGBA8_u32 primColor;
    /* 0x0C */ Color_RGBA8_u32 envColor;
} VisZbuf; // size = 0x10
```
These look like they have a common substruct.

Reply:
It's also set from the `PRIM_COLOR` registers:
```c
        sVisCvg.primColor.r = R_FB_FILTER_PRIM_COLOR(0);
        sVisCvg.primColor.g = R_FB_FILTER_PRIM_COLOR(1);
        sVisCvg.primColor.b = R_FB_FILTER_PRIM_COLOR(2);
```
Might want to consider changing to `PRIMCOLOR` to match the gbi convention too.